packages/grid/_modules_/grid/components/columnHeaders/GridColumnHeaderItem.tsx

```diff
@@ -135,13 +135,13 @@ export const GridColumnHeaderItem = React.memo(
     }
 
     const cssClasses = clsx(
-      column.headerAlign === 'center' && 'MuiDataGrid-columnHeaderCenter',
-      column.headerAlign === 'right' && 'MuiDataGrid-columnHeaderRight',
+      column.headerAlign === 'center' && 'MuiDataGrid-columnHeader--alignCenter',
+      column.headerAlign === 'right' && 'MuiDataGrid-columnHeader--alignRight',
       {
-        'MuiDataGrid-columnHeaderSortable': column.sortable,
-        'MuiDataGrid-columnHeaderMoving': isDragging,
-        'MuiDataGrid-columnHeaderSorted': isColumnSorted,
-        'MuiDataGrid-columnHeaderNumeric': isColumnNumeric,
+        'MuiDataGrid-columnHeader--sortable': column.sortable,
+        'MuiDataGrid-columnHeader--moving': isDragging,
+        'MuiDataGrid-columnHeader--sorted': isColumnSorted,
+        'MuiDataGrid-columnHeader--numeric': isColumnNumeric,
         'MuiDataGrid-withBorder': showColumnRightBorder,
```

Comment:
```suggestion
        'MuiDataGrid-columnHeader--withBorder': showColumnRightBorder,
```

Reply:
That is the same class as the one used for the cell. I don't think we need a duplicate for that one.